kong/kong.lua

```diff
@@ -252,6 +252,16 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    plugin.handler:rewrite()
```

Comment:
This currently runs the handler on every plugin loaded in the system. Say we now have some 25 plugins, so on every request we're now invoking 25 plugins. Not just configured ones, but all available ones.

As the rewrite phase does not have a consumer yet, and also doesn't have an api yet, the only plugins that should be selected here are the global ones.